src/components/ToggleRefinement/ToggleRefinement.tsx

```diff
@@ -0,0 +1,50 @@
+/** @jsx h */
+
+import { h } from 'preact';
+import {
+  ToggleRefinementRenderState,
+  ToggleRefinementValue,
+} from '../../connectors/toggle-refinement/connectToggleRefinement';
+import { PreparedTemplateProps } from '../../lib/utils/prepareTemplateProps';
+import {
+  ToggleRefinementTemplates,
+  ToggleRefinementComponentCSSClasses,
+} from '../../widgets/toggle-refinement/toggle-refinement';
+
+import Template from '../Template/Template';
+export type ToggleRefinementProps = {
+  currentRefinement: ToggleRefinementValue;
+  refine: ToggleRefinementRenderState['refine'];
+  cssClasses: ToggleRefinementComponentCSSClasses;
+  templateProps: PreparedTemplateProps<ToggleRefinementTemplates>;
+};
+
+const ToggleRefinement = ({
+  currentRefinement,
+  refine,
+  cssClasses,
+  templateProps,
+}: ToggleRefinementProps) => (
+  <div className={cssClasses.root}>
+    <label className={cssClasses.label}>
+      <input
+        className={cssClasses.checkbox}
+        type="checkbox"
+        checked={currentRefinement.isRefined}
+        onChange={event =>
+          refine({ isRefined: !(event.target as HTMLInputElement).checked })
```

Comment:
This can be a slight breaking change since previously the `refine` didn't look like this, but like [this](https://github.com/algolia/instantsearch.js/blob/master/src/widgets/toggle-refinement/toggle-refinement.js#L40:L40).
I do prefer this change of wrapping isRefined in an object consistently.
However it can be a breaking change to those who create custom components using the connector.
What do you think?

Oh, no. it's not a breaking change. Custom components will use the connector, not the widget or the component file. So it's only internal change 👍🏻